src/assets/js/access-groups.js

```diff
@@ -0,0 +1,326 @@
+    try {
+      return await this.api.execute({
+        path: 'selfInvitations/post',
+        endpointArguments: [groupName],
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async rejectInvitation(groupName) {
+    try {
+      return await this.api.execute({
+        path: 'selfInvitations/delete',
+        endpointArguments: [groupName],
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async getAccessGroupMemberInvitations(groupName) {
+    try {
+      return await this.api.execute({
+        path: 'groupInvitations/get',
+        endpointArguments: [groupName],
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async updateAccessGroupTOS(groupName, text) {
+    try {
+      return await this.api.execute({
+        path: 'terms/put',
+        endpointArguments: [groupName],
+        dataArguments: text,
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async addAccessGroupTOS(groupName, text) {
+    try {
+      return await this.api.execute({
+        path: 'terms/post',
+        endpointArguments: [groupName],
+        dataArguments: text,
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async deleteAccessGroupTOS(groupName) {
+    try {
+      return await this.api.execute({
+        path: 'terms/delete',
+        endpointArguments: [groupName],
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  async getAccessGroupTOS(groupName) {
+    try {
+      return await this.api.execute({
+        path: 'terms/get',
+        endpointArguments: [groupName],
+      });
+    } catch (e) {
+      console.log(e.message);
+      throw new Error(e.message);
+    }
+  }
+
+  // TODO: This method is one that will eventually be built but are not supported right now
+  async updateInviteText(text) {
+    return new Promise((res, rej) => {
+      res('text updated');
+    });
+  }
+
+  // TODO: This method is one that will eventually be built but are not supported right now
+  async resendInvitation(text) {
+    return new Promise((res, rej) => {
+      res('invitation resent');
+    });
+  }
+
+  static getAccessGroupUpdateData(data) {
+    console.log('getting data: ', data);
+    return {
+      typ: 'type' in data ? data.type : '',
+      description: 'description' in data ? data.description : '',
+      group_expiration: 'expiration' in data ? data.expiration : 0,
+    };
+  }
+
+  static async getUsers(q, groupName) {
+    try {
+      const api = new Api();
+      let users = await api.execute({
+        path: 'users/get',
+        endpointArguments: [q, groupName],
+      });
+      // TODO: Replace this with: users = users.filter(({ email }) => email !== null);
```

Comment:
We should not filter here. Username is always set for active users.
I think in general the rules is _first/last name_ →  _username_ → _uuid_ as a fallback chain.

Reply:
Fairplay.